hydra/test_utils/test_utils.py

```diff
@@ -277,8 +277,16 @@ def integration_test(
     filename: str = "task.py",
     env_override: Dict[str, str] = {},
     clean_environment: bool = False,
+    generate_custom_cmd: Callable = None,
 ) -> str:
-    Path(tmpdir).mkdir(parents=True, exist_ok=True)
+
+    modified_tmpdir = tmpdir
+    if task_config["tmpdir"]:
```

Comment:
This function is already taking a tmpdir.
Why do you need to pass another one via the config? can't you just pass the one you want as the existing tmpdir parameter?

Reply:
pytest is generating a unique tmpdir for each test.
If it is passed in, all tests will be using the same dir and will clash each other

Reply:
You can add a function to the tests to allow manipulating the tmp dir:

```python
class IntegrationTestSuite:
    def get_test_scrarch_dir(self, tmpdir : str) -> Optional[Path]:
      return tmpdir
```

And later in tests:
```python
    def test_custom_task_name(
        self,
        tmpdir: Path,
        task_config: DictConfig,
        overrides: List[str],
        filename: str,
        expected_name: str,
        task_launcher_cfg: DictConfig,
        extra_flags: List[str],
    ) -> None:
        ...
        integration_test(
            tmpdir=self.get_test_scrarch_dir(tmpdir),
            ...
         )
        
```